flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/common/utils/SchemaTransformTest.java

```diff
@@ -0,0 +1,231 @@
+                            .setMode("NULLABLE"),
+                    new TableFieldSchema()
+                            .setName("scion")
+                            .setType("RECORD")
+                            .setMode("NULLABLE")
+                            .setFields(subFields),
+                    new TableFieldSchema()
+                            .setName("associates")
+                            .setType("RECORD")
+                            .setMode("REPEATED")
+                            .setFields(subFields),
+                    new TableFieldSchema()
+                            .setName("geoPositions")
+                            .setType("GEOGRAPHY")
+                            .setMode("NULLABLE"));
+
+    @Test
+    public void testConvertBigQuerySchemaToAvroSchema() {
+        TableSchema tableSchema = new TableSchema();
+        tableSchema.setFields(fields);
+        Schema avroSchema =
+                SchemaTransform.toGenericAvroSchema("testSchema", tableSchema.getFields());
+
+        Assertions.assertThat(avroSchema.getField("number").schema())
+                .isEqualTo(Schema.create(Schema.Type.LONG));
+
+        Assertions.assertThat(avroSchema.getField("species").schema())
+                .isEqualTo(
+                        Schema.createUnion(
+                                Schema.create(Schema.Type.NULL),
+                                Schema.create(Schema.Type.STRING)));
+        Assertions.assertThat(avroSchema.getField("quality").schema())
+                .isEqualTo(
+                        Schema.createUnion(
+                                Schema.create(Schema.Type.NULL),
+                                Schema.create(Schema.Type.DOUBLE)));
+        Assertions.assertThat(avroSchema.getField("quantity").schema())
+                .isEqualTo(
+                        Schema.createUnion(
+                                Schema.create(Schema.Type.NULL), Schema.create(Schema.Type.LONG)));
+        Assertions.assertThat(avroSchema.getField("birthday").schema())
+                .isEqualTo(
+                        Schema.createUnion(
+                                Schema.create(Schema.Type.NULL),
+                                LogicalTypes.timestampMicros()
+                                        .addToSchema(Schema.create(Schema.Type.LONG))));
+        Assertions.assertThat(avroSchema.getField("birthdayMoney").schema())
+                .isEqualTo(
+                        Schema.createUnion(
+                                Schema.create(Schema.Type.NULL),
+                                LogicalTypes.decimal(38, 9)
+                                        .addToSchema(Schema.create(Schema.Type.BYTES))));
+        Assertions.assertThat(avroSchema.getField("lotteryWinnings").schema())
+                .isEqualTo(
+                        Schema.createUnion(
+                                Schema.create(Schema.Type.NULL),
+                                LogicalTypes.decimal(77, 38)
```

Comment:
Flink decimals are limited to precision of 38

Reply:
previous comment applies here as well.

Reply:
Same as mine - you cannot read a `BIGNUMERIC(76,38)` into flink